source/extensions/filters/http/adaptive_concurrency/controller/gradient_controller.cc

```diff
@@ -113,9 +113,11 @@ void GradientController::updateMinRTT() {
 
   {
     absl::MutexLock ml(&sample_mutation_mtx_);
-    min_rtt_ = processLatencySamplesAndClear();
-    stats_.min_rtt_msecs_.set(
-        std::chrono::duration_cast<std::chrono::milliseconds>(min_rtt_).count());
+    if (hist_sample_count(latency_sample_hist_.get()) != 0) {
+        min_rtt_ = processLatencySamplesAndClear();
+        stats_.min_rtt_msecs_.set(
+            std::chrono::duration_cast<std::chrono::milliseconds>(min_rtt_).count());
```

Comment:
Seems like a reasonable change.  Could you add a basic test to cover this behavior?

Reply:
My concern is that this doesn't address the root cause. I'd prefer a change that prevents this from ever happening and make this an assertion or error log. Thoughts?

Reply:
yeah, I'm fine if https://github.com/envoyproxy/envoy/pull/15008 could solve this min_rtt issue. But i wonder whether we could make the logic more robust, in case it is not the only cause.  
Like 
```
if (hist_sample_count(latency_sample_hist_.get()) != 0) {
....
} else {
  // error logs / stats
}
```
I think the question is more likely related to the convention of onboarding the new feature in envoy oss community: fail fast or fail safe?